bodhi/server/consumers/resultsdb.py

```diff
@@ -0,0 +1,87 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+The "resultsdb handler".
+
+This module is responsible for listening for messages from ResultsDB.
+If a message seems like it might be a result change for an update or
+build from an update, we re-check the gating decision for that update.
+"""
+
+import logging
+
+import fedora_messaging
+
+from bodhi.server.models import Build, TestGatingStatus, Update
+from bodhi.server.util import transactional_session_maker
+
+log = logging.getLogger(__name__)
+
+
+class ResultsdbHandler:
+    """
+    The Bodhi ResultsDB Handler.
+
+    A fedora-messaging listener waiting for messages from resultsdb that may
+    affect gating status for an update.
+    """
+
+    def __init__(self):
+        """Initialize the handler."""
+        self.db_factory = transactional_session_maker()
+
+    def __call__(self, message: fedora_messaging.api.Message):
+        """Handle messages arriving with the configured topic."""
+        msg = message.body
+        if not msg:
+            log.debug("Ignoring message without body.")
+            return
+
+        restype = msg.get("data", {}).get("type")
+
+        if restype is None:
+            log.error(f"Couldn't find result type in ResultsDB message {message.id}")
+            return
+        if restype not in ("koji_build", "bodhi_update"):
+            log.debug(f"Irrelevant result type {restype}")
+            return
+
+        passed = False
+        if msg["outcome"] in ("PASSED", "INFO"):
+            passed = True
+
+        with self.db_factory():
+            # find the update
+            if restype == "bodhi_update":
+                updateid = msg.get("data", {}).get("item")
+                if not updateid:
+                    log.error(f"Couldn't find update ID in message {message.id}")
+                    return
+                update = Update.get(updateid)
+                if not update:
+                    log.error(f"Couldn't find update {updateid} in DB")
+                    return
+            else:
+                nvr = msg["data"].get("nvr", msg["data"].get("item", ""))
+                if not nvr:
+                    log.error(f"Couldn't find nvr in message {message.id}")
+                    return
+                build = Build.get(nvr)
+                if not build:
+                    log.error(f"Couldn't find build {nvr} in DB")
+                    return
+                update = build.update
+
+            # update the gating status if there's a chance it changed
+            status = update.test_gating_status
+            if (
+                (passed and status == TestGatingStatus.passed) or
+                (not passed and status == TestGatingStatus.failed)
```

Comment:
Not important, but revisited Python style guide says operators should be preferably at the beginning of line.

```suggestion
                (passed and status == TestGatingStatus.passed)
                or (not passed and status == TestGatingStatus.failed)
```

https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator

Reply:
I've always hated that guideline. doesn't read right to me at all. so I tend to ignore it :P

Reply:
:+1: I often prefer operator in front but don't if it is elsewhere. It is the annoying linters that care. :)

Reply:
oh yeah, didn't notice flake was complaining.